server/src/test/java/org/cloudfoundry/identity/uaa/zone/JdbcIdentityZoneProvisioningTests.java

```diff
@@ -230,6 +231,34 @@ void testGetIdentityZone() {
         assertEquals(identityZone.getConfig().getTokenPolicy().getAccessTokenValidity(), retrievedIdZone.getConfig().getTokenPolicy().getAccessTokenValidity());
         assertEquals(identityZone.getConfig().getTokenPolicy().getRefreshTokenValidity(), retrievedIdZone.getConfig().getTokenPolicy().getRefreshTokenValidity());
         assertTrue(retrievedIdZone.isActive());
+
+        assertFalse(retrievedIdZone.getConfig().getLinks().getSelfService().isSelfServiceCreateAccountEnabled());
+        assertTrue(retrievedIdZone.getConfig().getLinks().getSelfService().isSelfServiceResetPasswordEnabled());
+        assertNull(retrievedIdZone.getConfig().getLinks().getSelfService().getPasswd());
+        assertNull(retrievedIdZone.getConfig().getLinks().getSelfService().getSignup());
+    }
+
+    @Test
+    void testGetIdentityZone_disabledSelfServiceLinks() {
+        IdentityZone identityZone = MultitenancyFixture.identityZone(randomValueStringGenerator.generate(), randomValueStringGenerator.generate());
+        identityZone.setId(randomValueStringGenerator.generate());
+
+        IdentityZoneConfiguration config = new IdentityZoneConfiguration();
+        config.getLinks().getSelfService().setSelfServiceLinksEnabled(true);
+        config.getLinks().getSelfService().setSignup("");
+        config.getLinks().getSelfService().setPasswd("/forgot_password");
+        identityZone.setConfig(config);
+
+        jdbcIdentityZoneProvisioning.create(identityZone);
+
+        IdentityZone retrievedIdZone = jdbcIdentityZoneProvisioning.retrieve(identityZone.getId());
+
+        assertEquals(identityZone.getId(), retrievedIdZone.getId());
+
+        assertFalse(retrievedIdZone.getConfig().getLinks().getSelfService().isSelfServiceCreateAccountEnabled());
+        assertTrue(retrievedIdZone.getConfig().getLinks().getSelfService().isSelfServiceResetPasswordEnabled());
+        assertEquals(retrievedIdZone.getConfig().getLinks().getSelfService().getPasswd(), "/forgot_password");
+        assertEquals(retrievedIdZone.getConfig().getLinks().getSelfService().getSignup(), "");
```

Comment:
[testCreateIdentityZone()](https://github.com/GESoftware-CF/uaa/blob/predix_extensions_75.18.1/server/src/test/java/org/cloudfoundry/identity/uaa/zone/JdbcIdentityZoneProvisioningTests.java#L58-L73) validates zone creation when no explicit config is provided.  Add validation that in this case create account is disabled, reset password is enabled, and both signup and passwd links are null.

Also, should add the following tests:
- _testCreateIdentityZone_enabledLegacySelfService()_: selfServiceLinksEnabled = true, signup = "", passwd = "/forgot_password" (use case of creating zone by service broker)
- _testCreateIdentityZone_enabledSelfServiceCreateAccount()_
- _testCreateIdentityZone_enabledSelfServiceResetPassword()_
- _testCreateIdentityZone_disabledSelfServiceResetPassword()_
- _testCreateIdentityZone_disabledSelfServiceCreateAccount()_

[testGetIdentityZone()](https://github.com/GESoftware-CF/uaa/blob/predix_extensions_75.18.1/server/src/test/java/org/cloudfoundry/identity/uaa/zone/JdbcIdentityZoneProvisioningTests.java#L218-L233) validates retrieving a zone with default config.  Add validation that in this case create account is disabled, reset password is enabled, and both signup and passwd links are null.

Also, add the following tests:
- _testGetIdentityZone_disabledLegacySelfService()_: selfServiceLinksEnabled = false, signup = null, passwd = null
- _testGetIdentityZone_enabledLegacySelfService()_: selfServiceLinksEnabled = true, signup = "", passwd = "/forgot_password"
- _testGetIdentityZone_enabledLegacySelfServiceAndLinks()_: selfServiceLinksEnabled = true, signup = "/create_account", passwd = "/forgot_password"

Reply:
Done